vms/platformvm/state/state.go

```diff
@@ -1135,28 +1139,35 @@ func (s *state) loadCurrentValidators() error {
 		}
 		tx, _, err := s.GetTx(txID)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed loading validator transaction txID %v, %w", txID, err)
+		}
+		stakerTx, ok := tx.Unsigned.(txs.Staker)
+		if !ok {
+			return fmt.Errorf("expected tx type txs.Staker but got %T", tx.Unsigned)
 		}
 
 		metadataBytes := validatorIt.Value()
+
+		defaultStartTime := stakerTx.StartTime()
 		metadata := &validatorMetadata{
 			txID: txID,
+			// use the start values as the fallback
+			// in case they are not stored in the database
 			// Note: we don't provide [LastUpdated] here because we expect it to
 			// always be present on disk.
+			StakerStartTime: defaultStartTime.Unix(),
 		}
 		if err := parseValidatorMetadata(metadataBytes, metadata); err != nil {
 			return err
 		}
 
-		stakerTx, ok := tx.Unsigned.(txs.Staker)
-		if !ok {
-			return fmt.Errorf("expected tx type txs.Staker but got %T", tx.Unsigned)
-		}
-
 		staker, err := NewCurrentStaker(txID, stakerTx, metadata.PotentialReward)
 		if err != nil {
 			return err
 		}
+		ShiftStakerAheadInPlace(staker, time.Unix(metadata.StakerStartTime, 0))
```

Comment:
note: in the following continuous staking PRs the NewCurrentStaker constructor will change slightly and some of these updates won't be necessary anymore. Here we chose to keep current NewCurrentStaker and carry out the updates to limit the number of changes